extensions/sql-bindings/src/test/common/azureFunctionsUtils.test.ts

```diff
@@ -349,13 +343,364 @@ describe('AzureFunctionUtils', function (): void {
+
+			should(quickPickStub.getCall(1).args).containDeepOrdered([
+				[constants.connectionProfile, constants.userConnectionString],
+				{
+					canPickMany: false,
+					title: constants.selectConnectionString,
+					ignoreFocusOut: true
+				}]
+			);
+			should(inputBoxStub.callCount).be.equal(2, 'showInputBox should have been called twice');
+			should(quickPickStub.callCount).be.equal(3, 'showQuickPick should have been called three times');
+			should(result?.connectionStringSettingName).be.equal(undefined, 'Should return undefined since user cancelled out of connection string setting name prompt');
+		});
+
+		it('Should prompt connection string method when user cancels out of manually entering connection string', async () => {
+			// create fake connection string settings for local.setting.json to be used
+			// getLocalSettingsJson stub
+			sinon.stub(fs.promises, 'access').resolves();
+			sinon.stub(fs, 'readFileSync').withArgs(sinon.match.any).returns(
+				`{"IsEncrypted": false,
+				"Values": {"test1": "test1", "test2": "test2", "test3":"test3"}}`
+			);
+
+			let quickPickStub = sinon.stub(vscode.window, 'showQuickPick').onFirstCall().resolves({ label: constants.createNewLocalAppSettingWithIcon }); // user chooses to create new connection string setting name
+			let inputBoxStub = sinon.stub(vscode.window, 'showInputBox').onFirstCall().resolves('testConnectionStringName'); // enter connection string setting name
+
+			// user chooses to manually enter connection string
+			quickPickStub.onSecondCall().resolves((constants.connectionProfile) as any);
+			sinon.stub(utils, 'getVscodeMssqlApi').resolves(testUtils.vscodeMssqlIExtension.object);
+			// user cancels out of connection profile prompt
+			testUtils.vscodeMssqlIExtension.setup(x => x.promptForConnection(true)).returns(() => Promise.resolve(undefined));
+
+			let result = await azureFunctionsUtils.promptAndUpdateConnectionStringSetting(fileUri);
+
+			should(quickPickStub.getCall(2).args).containDeepOrdered([
+				[constants.connectionProfile, constants.userConnectionString],
+				{
+					canPickMany: false,
+					title: constants.selectConnectionString,
+					ignoreFocusOut: true
+				}]
+			);
+			should(inputBoxStub.callCount).be.equal(1, 'showInputBox should have been called');
+			should(quickPickStub.callCount).be.equal(3, 'showQuickPick should have been called three times');
+			should(result?.connectionStringSettingName).be.equal(undefined, 'Should return undefined since user cancelled out of connection string setting name prompt');
+		});
+
+		it('Should prompt connection string settings when user cancels out of manually entering connection string name prompt if connection string exist in local.settings.json', async () => {
+			// create fake connection string settings for local.setting.json to be used
+			// getLocalSettingsJson stub
+			sinon.stub(fs.promises, 'access').resolves();
```

Comment:
Consider : 

1. Making a helper function for these setup calls (doesn't save much, but even just quickly being able to tell that these all share setup logic is useful)
2. or putting all the ones that share a setup into another sub-suite and use beforeEach